### PR TITLE
Fixing the bugfix #1718

### LINKED ISSUE #1717: Dynamic Forms with TaxonomyFieldTypeMulti fields don't show
#### Category
[ ] Enhancement

[X] Bug

[ ] Question

#### Version

Please specify what version of the library you are using: [3.16.0]

#### Expected / Desired Behavior / Question
The fix I made for issue #1568 did not cover all cases and including a TaxonomyFieldTypeMulti field results in the Dynamic Form not stopping to load.

### PR DESCRIPTION
**Pull request by IRRDC**
Bugfix #1718 for bug #1717 was faulty and gets fixed with this.

| Q               | A
| --------------- | ---
| Bug fix?        | [X]
| New feature?    | [ ]
| New sample?      | [ ]
| Related issues?  | fixes #1717 

#### What's in this Pull Request?

Removed the wrongly added reference to "results".

### CODE REVIEW

File: src/controls/dynamicForm/DynamicForm.tsx [hunk 1]
```diff
@@ -597,17 +597,17 @@ export class DynamicForm extends React.Component<
...
                 selectedTags.push({
                   key: element.TermGuid,
                   name: element.Label,
                 });
               });
 
               defaultValue = selectedTags;
-            } else if (defaultValue && defaultValue.results) {
-              defaultValue.results.forEach((element) => {
+            } else if (defaultValue) {
```
Comment (michaelmaillot) [inline src/controls/dynamicForm/DynamicForm.tsx:609]: Lines 609-616 can also be removed (as default values are handled below), they raised an error when loading a new item with default terms configured on the field.

**Review by michaelmaillot — CHANGES_REQUESTED** [review]
Thanks for this PR! I've add a small change request.

**Comment by IRRDC** [comment]
@michaelmaillot Thank you for noticing that. I have updated the code.

**Comment by michaelmaillot** [comment]
Sorry @IRRDC but as this is a hotfix release, target branch will be dev and not main (then we will cherry-pick your fix to master in order to release a patch verison).

Would you mind resolving conflicts before the merge?

**Comment by IRRDC** [comment]
@michaelmaillot should be resolved now.

**Comment by michaelmaillot** [comment]
Sorry @IRRDC but old code is still present [here](https://github.com/pnp/sp-dev-fx-controls-react/pull/1736/commits/0065dab67da94010e7d049de7da528f4c343fdb5#diff-0cf968dc2eb61b72577675d2abf5f042e96e879454818cfa388211bfcb43ecdcR1204)

**Comment by IRRDC** [comment]
@michaelmaillot Looks like the merge removed all my changes. I have restored them now.

**Review by michaelmaillot — APPROVED** [review]

**Pull request merged**